repoaudit/repoaudit/__init__.py

```diff
@@ -60,32 +92,41 @@ def main() -> None:
 
 @main.command()
 @recursive_option
-@click.argument("url")
-@click.option("--dists", help="Comma separated list of distributions.")
+@click.argument("url", required=False)
+@click.option("--dists", required=False, help="Comma separated list of distributions.")
+@click.option(
+    "--apt-source",
+    "-a",
+    required=False,
+    type=click.File("r"),
+    help=(
+        "Supply apt sources.list file. When provided, all entries"
+        "will be parsed for repo urls and their respective dists."
+    )
+)
 @file_option
 @pubkey_option
-def apt(recursive: bool, url: str, dists: str, output: str, pubkeys: str) -> None:
+def apt(recursive: bool, url: str, dists: str, apt_source: str, output: str, pubkeys: str) -> None:
     """Validate an apt repository at URL."""
-    if recursive:
-        urls = get_repo_urls(url)
-    else:
-        urls = [url]
-
-    if dists:
-        dist_set = set(dists.split(","))
-    else:
-        dist_set = None
-
-    errors = RepoErrors()
-
-    with _gpg_cmdline(pubkeys) as gpg:
+    if apt_source:
         try:
-            for repo_url in urls:
-                check_apt_repo(repo_url, dist_set, gpg, errors)
-        except KeyboardInterrupt:
-            pass
-
-    output_result(errors, output)
+            lines = [line.strip() for line in apt_source if line.startswith("deb")]
+            for line in lines:
+                fields = line.split(" ")
+                index: int
+                if (fields[1].startswith("http")):
```

Comment:
One other thing to consider, `sources.list` supports "comments." This would be any line where the first non-space character is `#`. 
There can also be empty lines in a `sources.list` file. We should ignore those entirely.